### ISSUE TITLE
predict_instance_segmentation

### ISSUE THREAD

**Issue by BIT-MJY** [EdwardLeeLPZ/PowerBEV#1]
Hello @EdwardLeeLPZ ,

Thanks for your great work! I have one question about the `predict_instance_segmentation` function in instance.py. Could you please tell me why you use `output['instance_flow'][b, 1:2].detach(),` rather than the first predicted instance flow [b, 0:1] to generate the instance in `get_instance_segmentation_and_centers`?

**Comment by EdwardLeeLPZ** [comment]
> Hello @EdwardLeeLPZ ,
> 
> Thanks for your great work! I have one question about the `predict_instance_segmentation` function in instance.py. Could you please tell me why you use `output['instance_flow'][b, 1:2].detach(),` rather than the first predicted instance flow [b, 0:1] to generate the instance in `get_instance_segmentation_and_centers`?

Hello @BIT-MJY ,
Your question is critical. To better understand the different outputs used for post-processing, please refer to this figure:
![image](https://github.com/EdwardLeeLPZ/PowerBEV/assets/49630295/40326925-ecc8-4bfb-8bc9-cf0bddc85998)
The `get_instance_segmentation_and_centers` function is used for giving an initial instance segmentation of the first valid prediction frame (t=0). It is implemented in the following manner:

1. Extract local maxima from `torch.softmax(output['segmentation'], dim=2)[b, 0:1, vehicles_id].detach()` as the centers of instances at the past moment (t=-1) [(Line 44 - 55)](https://github.com/EdwardLeeLPZ/PowerBEV/blob/master/powerbev/utils/instance.py#L44-L55);
2. Use backward centripetal flow `output['instance_flow'][b, 1:2].detach()` to find the corresponding position in the past frame (t=-1) to which all pixels at the moment t=0 are pointing [(Line 59- 72)](https://github.com/EdwardLeeLPZ/PowerBEV/blob/master/powerbev/utils/instance.py#L59-L72);
3. Compare the above positions being pointed to (t=-1) with the predicted centers of the past frame (t=-1), and assign the nearest center to each present pixel (t=0), i.e., the identification [(Line 75- 77)](https://github.com/EdwardLeeLPZ/PowerBEV/blob/master/powerbev/utils/instance.py#L75-L77);
4. Filter with `foreground_masks[b, 1:2].detach()` at moment t=0 to avoid background pixels being assigned as instances [(Line 103)](https://github.com/EdwardLeeLPZ/PowerBEV/blob/master/powerbev/utils/instance.py#L103).

It is worth noting that this step only assigns IDs to the current instances (t=0), and the centers of the past frame (t=-1) are only used as assistance, **not part of the final valid prediction outputs (t=0, 1, 2, ...)**.

I hope I have answered your question clearly, if not, feel free to raise your question further.
